src/flowgger/fuzzer.rs

```diff
@@ -0,0 +1,210 @@
+    pub fn update_file_rotation_defaults_in_config(config: &mut Config){
+        if let Some(entry) = config.config.get_mut("output").unwrap().get_mut("file_rotation_size"){
+            *entry = Value::Integer(0);
+        }
+
+        if let Some(entry) = config.config.get_mut("output").unwrap().get_mut("file_rotation_time"){
+            *entry = Value::Integer(0);
+        }
+    }
+
+    pub fn remove_output_file(file_output_path: &str){
+        let _ = fs::remove_file(file_output_path);
+    }
+    
+    // Initialize the SyncSender which will be used to send data to the output file
+    fn initialize_sender(tx: SyncSender<Vec<u8>>){
+        let mut sender_guard = SYNC_SENDER.lock().unwrap();
+        if sender_guard.is_none(){
+            *sender_guard = Some(tx);
+        }
+
+        drop(sender_guard);
+    }
+
+    /// Start an input listener which writes data to the output file once received.
+    pub fn start_file_output(config: &Config, rx: Receiver<Vec<u8>>){
+
+        let output_format = config
+            .lookup("output.format")
+            .map_or(DEFAULT_OUTPUT_FORMAT, |x| {
+                x.as_str().expect("output.format must be a string")
+            });
+
+        let output = get_file_output(&config);
+        let output_type = config
+            .lookup("output.type")
+            .map_or(DEFAULT_OUTPUT_TYPE, |x| {
+                x.as_str().expect("output.type must be a string")
+            });
+
+        let _output_framing = match config.lookup("output.framing") {
+            Some(framing) => framing.as_str().expect("output.framing must be a string"),
+            None => match (output_format, output_type) {
+                ("capnp", _) | (_, "kafka") => "noop",
+                (_, "debug") | ("ltsv", _) => "line",
+                ("gelf", _) => "nul",
+                _ => DEFAULT_OUTPUT_FRAMING,
+            },
+        };
+        let merger: Option<Box<dyn Merger>> = Some(Box::new(LineMerger::new(&config)) as Box<dyn Merger>);
+
+        let arx = Arc::new(Mutex::new(rx));
+        output.start(arx, merger);
+    }
+
+    fn get_file_output(config: &Config) -> Box<dyn Output> {
+        Box::new(FileOutput::new(config)) as Box<dyn Output>
+    }
+    
+    pub fn fuzz_target_rfc3164(data: String) {
+        let config = STATIC_CONFIG.clone();
+        
+        let mut sender_guard = match SYNC_SENDER.lock() {
+            Ok(guard) => guard,
+            Err(_poisoned_error) => {
+                // Handle the poisoned Mutex
+                let guard = _poisoned_error.into_inner();
+                guard
+            }
+        };
+        let tx: &mut SyncSender<Vec<u8>> = sender_guard.as_mut().unwrap();
+
+        let encoder = get_encoder_rfc3164(&config);
+        let decoder = get_decoder_rfc3164(&config);
+        let (decoder, encoder): (Box<dyn Decoder>, Box<dyn Encoder>) =
+            (decoder.clone_boxed(), encoder.clone_boxed());
```

Comment:
i think we talked that basically only the handle_record needs to be called fore very message. cloning the static config and recreating the same encoder and decoder for every message is useless.

I expect tx, decoder, and encoder to be shared. E.g. have a structure containing the 3 of them which you chsare with your mutex ?